pkg/webhook/virtualmachine/virtualmachine.go

```diff
@@ -137,16 +146,17 @@ func patchChange(pathChange string, original, current interface{}) ([]jsonpatch.
 }
 
 // mutateCreateVirtualMachinesFn calls the create allocation function
-func (a *virtualMachineAnnotator) mutateCreateVirtualMachinesFn(ctx context.Context, virtualMachine *kubevirt.VirtualMachine, parentLogger logr.Logger) error {
+func (a *virtualMachineAnnotator) mutateCreateVirtualMachinesFn(virtualMachine *kubevirt.VirtualMachine, transactionTimestamp string, parentLogger logr.Logger) error {
 	logger := parentLogger.WithName("mutateCreateVirtualMachinesFn")
 	logger.Info("got a create mutate virtual machine event")
+	pool_manager.SetTransactionTimestampAnnotationToVm(virtualMachine, transactionTimestamp)
 
 	existingVirtualMachine := &kubevirt.VirtualMachine{}
 	err := a.client.Get(context.TODO(), client.ObjectKey{Namespace: virtualMachine.Namespace, Name: virtualMachine.Name}, existingVirtualMachine)
 	if err != nil {
 		// If the VM does not exist yet, allocate a new MAC address
 		if apierrors.IsNotFound(err) {
-			if virtualMachine.ObjectMeta.DeletionTimestamp.IsZero() {
+			if pool_manager.IsVirtualMachineNotMarkedForDeletion(virtualMachine) {
 				logger.V(1).Info("The VM is not being deleted.")
 				// If the object is not being deleted, then lets allocate macs and add the finalizer
 				err = a.poolManager.AllocateVirtualMachineMac(virtualMachine, logger)
```

Comment:
When we add the entry to the cache we are "suspending" the transaction and and controller will resume it.
```golang
transaction.Suspend(poolManager)
```